uid/src/main/java/com/tomtom/kotlin/uid/Uid.kt

```diff
@@ -0,0 +1,225 @@
+        if (uuid != other.uuid) return false
+        return true
+    }
+
+    override fun hashCode(): Int {
+        return uuid.hashCode()
+    }
+
+    /**
+     * Returns string representation of this Uid. Opposite of [fromString].
+     *
+     * @return String representation of id.
+     */
+    override fun toString(): String {
+        return uuid
+    }
+
+    companion object {
+        private const val serialVersionUID = 1L
+        private const val UUID_DASH = '-'
+        private const val UUID_MIN_LENGTH = 9
+        private const val UUID_MAX_LENGTH = 36
+        private val UUID_DASH_POS = intArrayOf(8, 13, 18, 23)
+
+        /**
+         * Return whether an ID is a valid UUID.
+         *
+         * @param id String representation of UUID.
+         * @return True if this is a valid UUID. The construction of a Uid() from this id
+         * is guaranteed succeed in that case.
+         */
+        fun isValid(id: String?): Boolean {
+            return if (id == null) {
+                false
+            } else try {
+                Uid<Any>(id)
+                true
+            } catch (ignored: IllegalArgumentException) {
+                false
+            }
+        }
+
+        /**
+         * Instantiates a Uid with given id. Mainly used for deserialization.
+         * Opposite of [toString].
+         *
+         * @param <T> Uid type.
+         * @param id  String representation of id.
+         * @return Uid.
+         * @throws IllegalArgumentException If name does not conform to the string
+         * representation as described in [UUID.toString]. Use [isValid] to make sure the
+         * string is valid.
+        </T> */
+        fun <T> fromString(id: String): Uid<T> {
+            return Uid<T>(id)
+        }
+
+        /**
+         * Instantiates a Uid with given id. Opposite of [toHexString].
+         *
+         * @param <T> Uid type.
+         * @param id  Hex string representation of id, must be exactly 32 characters long.
+         * @return Uid.
+         * @throws IllegalArgumentException If name does not conform to the string
+         * representation as described in [UUID.toString]. Use [isValid] to make sure the
+         * string is valid.
+        </T> */
+        fun <T> fromHexString(id: String): Uid<T> {
+            require(id.length == 32)
+            val msb: Long =
+                id.substring(0, 8).toLong(16) shl 32 or id.substring(8, 16).toLong(16)
+            val lsb: Long =
+                id.substring(16, 24).toLong(16) shl 32 or id.substring(24, 32).toLong(16)
+            return Uid<T>(UUID(msb, lsb))
+        }
+
+        /**
+         * Return if string contains valid UUID characters only.
+         * Must be converted to lowercase already.
+         *
+         * @param uuid Input UUID. Should be lowercase already.
+         * @return True if valid characters only.
+         */
+        private fun onlyContainsValidUUIDCharacters(uuid: String): Boolean {
```

Comment:
Not sure we wanna stick to IVI (and AOSP) styling guidelines, but we normally [treat acronyms as words](https://source.android.com/setup/contribute/code-style#treat-acronyms-as-words) (even when a non-compliant lib already violates that).
```suggestion
        private fun onlyContainsValidUuidCharacters(uuid: String): Boolean {
```

Reply:
Changed.